### DISCUSSION TITLE
Support for other shared types

### DISCUSSION THREAD

**Discussion by pattobrien** [cachapa/crdt#17] [General]
Thanks for the package! I'm curious if you had any thoughts on adding other Shared Types, like Lists, Text, etc. I'd be happy to help contribute where possible as well!

**Comment by cachapa** [comment]
The reason we stick to very basic values is because this package is meant to be implemented by various different backend technologies like no-sql, sql, flat maps, etc.

The more specialized types we add, the larger the surface for potential incompatibility down the line is introduced.

For example, Lists are difficult to represent in SQL engines, and Sqlite doesn't support DateTime types.

I'm also not sure what you mean by Text type - Strings are already supported and they're actually the most common fallback to represent all other types, e.g. DateTime.
